src/lib.rs

```diff
@@ -1,13 +1,93 @@
-use zonebuilder;
-use pyo3::prelude::*;
+#[allow(unused)]
+use geo::{map_coords::MapCoordsInplace, Point};
+use pyo3::{prelude::*, types::PyDict, Python};
+use pythonize::pythonize;
+use zonebuilder::{self, Params};
 
 #[pyfunction]
 fn triangular_sequence(n: usize) -> Vec<f64> {
     zonebuilder::triangular_sequence(n)
 }
 
+#[pyclass]
+#[derive(Clone)]
+struct Pyparams {
+    num_segments: usize,
+    distances: Vec<f64>,
+    num_vertices_arc: usize,
+    precision: usize,
+    projected: bool,
+}
+impl Default for Pyparams {
+    fn default() -> Self {
+        Pyparams {
+            num_segments: 12,
+            distances: triangular_sequence(5),
+            num_vertices_arc: 10,
+            precision: 6,
+            projected: false,
+        }
+    }
+}
+impl Pyparams {
+    fn to_zb_params(&self) -> Params {
+        Params {
+            num_segments: self.num_segments,
+            distances: self.distances.clone(),
+            num_vertices_arc: self.num_vertices_arc,
+            precision: self.precision,
+            projected: self.projected,
+        }
+    }
+}
+
+#[pyfunction(kwargs = "**")]
+fn kwargsparse(kwargs: Option<&PyDict>) -> PyResult<Pyparams> {
```

Comment:
Whew, this kind of boilerplate could really add up. I'd be surprised if someone doesn't create a macro or some other approach eventually

Reply:
I haven't used macros before but I can look into it.

Reply:
I wouldn't worry about it for this PR. I'm saying that I would expect `pyo3` to eventually provide an easier way to do this automatically

Reply:
Ah got it. Thanks for the feedback!